### ISSUE TITLE
Determinisic training dosen't work

### ISSUE THREAD

**Issue by GYDDHPY** [MIC-DKFZ/nnUNet#1218]
The deterministic option provided by nnUNet seems failed to ensure reproduceable training. 
Even I set the deterministic option to be true, the training loss of epochs can be different if I run the training again.

These are random settings I used.
```
np.random.seed(12345)  
torch.manual_seed(12345)
if torch.cuda.is_available():
  torch.cuda.manual_seed_all(12345)
cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
random.seed(1234567)
```
Is there anyone knows how to guarantee the nnUNet training procedure reproduceable ?

**Comment by nickl1234567** [comment]
I think the get_moreDA_augmentation function introduces stochasticity (e.g.: https://github.com/MIC-DKFZ/nnUNet/blob/aa53b3b87130ad78f0a28e6169a83215d708d659/nnunet/training/network_training/nnUNetTrainerV2.py#L105). The function does already provide parameters for seed values. 
I am not sure, if there is another source of randomness...

**Comment by nickl1234567** [comment]
In my version of the nnUNet I am using an old version version of batchgenerators, that had a bug with reproducibility. I did not test it with a new version, but it should be fixed now.
See:
https://github.com/MIC-DKFZ/batchgenerators/issues/17
